lib/logidze/versioned_association.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'active_support'
+
+module Logidze
+  module VersionedAssociation
+    module SingularAssociation
+      def reader(force_reload = false)
+        super_result = super
+        return super_result unless owner.logidze_requested_ts
+
+        logidze_modify_target(target, owner.logidze_requested_ts)
+      end
+    end
+
+    module CollectionAssociation
+      def load_target
```

Comment:
In case of collection association we use `load_target`, because it will be used by `CollectionProxy` object